### PR TITLE
docs: update CONTRIBUTING.md build-env setup section & add TESTING.md

### PR DESCRIPTION
**Pull request by sarmahaj**
This PR contains
- update CONTRIBUTING.md file to provide detailed information on setting
     up the build environment on Linux/mac machine.
- create a new file 'TESTING.md' to provide information on testing fdo
    e2e scenario.

### CODE REVIEW

**Comment by 7flying** [comment]
Looking great!
We only need the commit about the docs, please remove the other two.

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -20,5 +20,13 @@
 		"tamasfe.even-better-toml",
 		"serayuzgur.crates"
 	],
-	"remoteUser": "vscode"
+	"hostRequirements": {
+		"memory": "4gb"
+	},
+	"remoteUser": "vscode",
```
Comment (nullr0ute) [inline .devcontainer/devcontainer.json:26]: Should this be vscode specific?

File: admin-tool/Cargo.toml [hunk 2]
```diff
@@ -8,7 +8,7 @@ edition = "2018"
 
 [dependencies]
 anyhow = "1"
-openssl = "0.10"
+openssl = "0.10.35"
```
Comment (nullr0ute) [inline admin-tool/Cargo.toml:11]: These should be split out to a different commit as they're an unrelated change

Comment (nullr0ute) [reply admin-tool/Cargo.toml:11]: In fact these changes are already upstream, I think you may need to rebase

Comment (sarmahaj) [reply admin-tool/Cargo.toml:11]: my commit contains only two file changes https://github.com/fedora-iot/fido-device-onboard-rs/commit/7393c5f3852605a63bf55a0b9995700657ccfd16 . CONTRIBUTING.md and TESTING.md . above two changes are earlier ,I guess.

Comment (nullr0ute) [reply admin-tool/Cargo.toml]: Please rebase your branch and re-push, that should resolve it

**Review by nullr0ute — APPROVED** [review]
Looks great,  thanks
